Build failure on 32 bit architectures: error: conversion from 'hsize_t' {aka 'long  long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]

**Describe the bug**
While building HighFive for inclusion in [NeuroFedora](https://pagure.io/neuro-sig/NeuroFedora/issue/440), we're seeing builds fail on 32 bit architectures: i686 and armv7hl

**To Reproduce**
Steps to reproduce the behavior:
Attempt to build on a 32 bit platform

**Expected behavior**
Build should pass (or if 32 bit architectures aren't supported, this should be documented)

**Stacktrace**
```
[ 19%] Building CXX object src/examples/CMakeFiles/boost_ublas_double_bin.dir/boost_ublas_double.cpp.o
cd /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/i686-redhat-linux-gnu/src/examples && /usr/bin/g++ -DBOOST_ALL_NO_LIB -DH5_USE_BOOST -DH5_USE_EIGEN -DH5_USE_XTENSOR -DMPI_NO_CPPBIND -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include -isystem /usr/include/eigen3 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -g  -Wall -Wextra -Wno-unused-parameter  -Werror -Wshadow -Wnon-virtual-dtor -Wunused -Woverloaded-virtual -Wformat=2 -Wconversion -Wsign-conversion -Wno-error=deprecated-declarations -Wpedantic -Wcast-align -Wdouble-promotion -O2 -std=c++14 -o CMakeFiles/boost_multi_array_2D_bin.dir/boost_multi_array_2D.cpp.o -c /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/boost_multi_array_2D.cpp
cd /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/i686-redhat-linux-gnu/src/examples && /usr/bin/g++ -DBOOST_ALL_NO_LIB -DH5_USE_BOOST -DH5_USE_EIGEN -DH5_USE_XTENSOR -DMPI_NO_CPPBIND -D_FILE_OFFSET_BITS=64 -I/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include -isystem /usr/include/eigen3 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -g  -Wall -Wextra -Wno-unused-parameter  -Werror -Wshadow -Wnon-virtual-dtor -Wunused -Woverloaded-virtual -Wformat=2 -Wconversion -Wsign-conversion -Wno-error=deprecated-declarations -Wpedantic -Wcast-align -Wdouble-promotion -O2 -std=c++14 -o CMakeFiles/boost_ublas_double_bin.dir/boost_ublas_double.cpp.o -c /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/boost_ublas_double.cpp
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../H5Reference.hpp:80,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Converter_misc.hpp:30,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:25,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/select_by_id_dataset_cpp11.cpp:14:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp:23:52: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   23 |         return H5Iget_name(object.getId(), buffer, length); });
      |                                                    ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/select_by_id_dataset_cpp11.cpp:14:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:32:34: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   32 |         return H5Aget_name(_hid, length, buffer);
      |                                  ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../H5Reference.hpp:80,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Converter_misc.hpp:30,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:25,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/boost_ublas_double.cpp:19:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp:23:52: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   23 |         return H5Iget_name(object.getId(), buffer, length); });
      |                                                    ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../H5Reference.hpp:80,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Converter_misc.hpp:30,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:25,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/create_extensible_dataset.cpp:15:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp:23:52: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   23 |         return H5Iget_name(object.getId(), buffer, length); });
      |                                                    ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:90,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/select_by_id_dataset_cpp11.cpp:14:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp:81:49: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   81 |             return H5Fget_name(getId(), buffer, length);
      |                                                 ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../H5Reference.hpp:80,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Converter_misc.hpp:30,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:25,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/select_partial_dataset_cpp11.cpp:14:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp:23:52: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   23 |         return H5Iget_name(object.getId(), buffer, length); });
      |                                                    ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../H5Reference.hpp:80,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Converter_misc.hpp:30,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:25,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/renaming_objects.cpp:1:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp:23:52: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   23 |         return H5Iget_name(object.getId(), buffer, length); });
      |                                                    ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../H5Reference.hpp:80,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Converter_misc.hpp:30,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:25,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/boost_multi_array_2D.cpp:15:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp:23:52: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   23 |         return H5Iget_name(object.getId(), buffer, length); });
      |                                                    ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/create_extensible_dataset.cpp:15:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:32:34: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   32 |         return H5Aget_name(_hid, length, buffer);
      |                                  ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/renaming_objects.cpp:1:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:32:34: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   32 |         return H5Aget_name(_hid, length, buffer);
      |                                  ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/boost_ublas_double.cpp:19:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:32:34: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   32 |         return H5Aget_name(_hid, length, buffer);
      |                                  ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../H5Reference.hpp:80,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Converter_misc.hpp:30,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:25,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/read_write_single_scalar.cpp:13:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp:23:52: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   23 |         return H5Iget_name(object.getId(), buffer, length); });
      |                                                    ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:90,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/create_extensible_dataset.cpp:15:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp:81:49: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   81 |             return H5Fget_name(getId(), buffer, length);
      |                                                 ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:90,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/renaming_objects.cpp:1:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp:81:49: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   81 |             return H5Fget_name(getId(), buffer, length);
      |                                                 ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/boost_multi_array_2D.cpp:15:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:32:34: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   32 |         return H5Aget_name(_hid, length, buffer);
      |                                  ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:90,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/boost_ublas_double.cpp:19:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp:81:49: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   81 |             return H5Fget_name(getId(), buffer, length);
      |                                                 ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/select_partial_dataset_cpp11.cpp:14:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:32:34: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   32 |         return H5Aget_name(_hid, length, buffer);
      |                                  ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/read_write_single_scalar.cpp:13:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:32:34: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   32 |         return H5Aget_name(_hid, length, buffer);
      |                                  ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:90,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/boost_multi_array_2D.cpp:15:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp:81:49: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   81 |             return H5Fget_name(getId(), buffer, length);
      |                                                 ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:90,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/select_partial_dataset_cpp11.cpp:14:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp:81:49: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   81 |             return H5Fget_name(getId(), buffer, length);
      |                                                 ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:90,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/read_write_single_scalar.cpp:13:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp:81:49: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   81 |             return H5Fget_name(getId(), buffer, length);
      |                                                 ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../H5Reference.hpp:80,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Converter_misc.hpp:30,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:25,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/read_write_vector_dataset_references.cpp:15:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/../bits/H5Reference_misc.hpp:23:52: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   23 |         return H5Iget_name(object.getId(), buffer, length); });
      |                                                    ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:92,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/renaming_objects.cpp:1:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Path_traits_misc.hpp: In instantiation of 'std::string HighFive::PathTraits<Derivate>::getPath() const [with Derivate = HighFive::Group; std::string = std::__cxx11::basic_string<char>]':
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/renaming_objects.cpp:37:49:   required from here
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Path_traits_misc.hpp:39:81: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   39 |         return H5Iget_name(static_cast<const Derivate*>(this)->getId(), buffer, length);
      |                                                                                 ^~~~~~
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Path_traits_misc.hpp: In instantiation of 'std::string HighFive::PathTraits<Derivate>::getPath() const [with Derivate = HighFive::DataSet; std::string = std::__cxx11::basic_string<char>]':
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/renaming_objects.cpp:38:53:   required from here
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Path_traits_misc.hpp:39:81: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:89,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/read_write_vector_dataset_references.cpp:15:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5Attribute_misc.hpp:32:34: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   32 |         return H5Aget_name(_hid, length, buffer);
      |                                  ^~~~~~
In file included from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/H5File.hpp:90,
                 from /builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/src/examples/read_write_vector_dataset_references.cpp:15:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp: In lambda function:
/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/include/highfive/bits/H5File_misc.hpp:81:49: error: conversion from 'hsize_t' {aka 'long long unsigned int'} to 'size_t' {aka 'unsigned int'} may change value [-Werror=conversion]
   81 |             return H5Fget_name(getId(), buffer, length);
      |                                                 ^~~~~~
cc1plus: all warnings being treated as errors
gmake[2]: Leaving directory '/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/i686-redhat-linux-gnu'
gmake[2]: *** [src/examples/CMakeFiles/boost_multi_array_2D_bin.dir/build.make:85: src/examples/CMakeFiles/boost_multi_array_2D_bin.dir/boost_multi_array_2D.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:304: src/examples/CMakeFiles/boost_multi_array_2D_bin.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
cc1plus: all warnings being treated as errors
cc1plus: all warnings being treated as errors
gmake[2]: Leaving directory '/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/i686-redhat-linux-gnu'
gmake[2]: *** [src/examples/CMakeFiles/read_write_single_scalar_bin.dir/build.make:85: src/examples/CMakeFiles/read_write_single_scalar_bin.dir/read_write_single_scalar.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:331: src/examples/CMakeFiles/read_write_single_scalar_bin.dir/all] Error 2
gmake[2]: *** [src/examples/CMakeFiles/boost_ublas_double_bin.dir/build.make:85: src/examples/CMakeFiles/boost_ublas_double_bin.dir/boost_ublas_double.cpp.o] Error 1
gmake[2]: Leaving directory '/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/i686-redhat-linux-gnu'
gmake[1]: *** [CMakeFiles/Makefile2:358: src/examples/CMakeFiles/boost_ublas_double_bin.dir/all] Error 2
cc1plus: all warnings being treated as errors
gmake[2]: *** [src/examples/CMakeFiles/renaming_objects_bin.dir/build.make:85: src/examples/CMakeFiles/renaming_objects_bin.dir/renaming_objects.cpp.o] Error 1
gmake[2]: Leaving directory '/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/i686-redhat-linux-gnu'
gmake[1]: *** [CMakeFiles/Makefile2:250: src/examples/CMakeFiles/renaming_objects_bin.dir/all] Error 2
cc1plus: all warnings being treated as errors
gmake[2]: *** [src/examples/CMakeFiles/select_by_id_dataset_cpp11_bin.dir/build.make:85: src/examples/CMakeFiles/select_by_id_dataset_cpp11_bin.dir/select_by_id_dataset_cpp11.cpp.o] Error 1
gmake[2]: Leaving directory '/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/i686-redhat-linux-gnu'
gmake[1]: *** [CMakeFiles/Makefile2:223: src/examples/CMakeFiles/select_by_id_dataset_cpp11_bin.dir/all] Error 2
cc1plus: all warnings being treated as errors
cc1plus: all warnings being treated as errors
gmake[2]: *** [src/examples/CMakeFiles/create_extensible_dataset_bin.dir/build.make:85: src/examples/CMakeFiles/create_extensible_dataset_bin.dir/create_extensible_dataset.cpp.o] Error 1
gmake[2]: Leaving directory '/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/i686-redhat-linux-gnu'
gmake[1]: *** [CMakeFiles/Makefile2:385: src/examples/CMakeFiles/create_extensible_dataset_bin.dir/all] Error 2
gmake[2]: *** [src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/build.make:85: src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/select_partial_dataset_cpp11.cpp.o] Error 1
gmake[2]: Leaving directory '/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/i686-redhat-linux-gnu'
gmake[1]: *** [CMakeFiles/Makefile2:196: src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/all] Error 2
cc1plus: all warnings being treated as errors
gmake[2]: *** [src/examples/CMakeFiles/read_write_vector_dataset_references_bin.dir/build.make:85: src/examples/CMakeFiles/read_write_vector_dataset_references_bin.dir/read_write_vector_dataset_references.cpp.o] Error 1
gmake[2]: Leaving directory '/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/i686-redhat-linux-gnu'
gmake[1]: Leaving directory '/builddir/build/BUILD/HighFive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d/i686-redhat-linux-gnu'
gmake[1]: *** [CMakeFiles/Makefile2:277: src/examples/CMakeFiles/read_write_vector_dataset_references_bin.dir/all] Error 2
gmake: *** [Makefile:163: all] Error 2

```

Complete build log attached:
[highfive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d-build-log.txt](https://github.com/BlueBrain/HighFive/files/6329314/highfive-b8998c6b90f3b1411f4c91f0c5491d738a5f611d-build-log.txt)


**Desktop (please complete the following information):**
 - OS: Fedora 35/rawhide
 - Version: tested with both version 2.2.2 and commit b8998c6b90f3b1411f4c91f0c5491d738a5f611d

**Additional context**
Here's the test build on the Fedora builders where only armv7hl and i686 fail:

https://koji.fedoraproject.org/koji/taskinfo?taskID=66124517

--- a/include/highfive/bits/H5Attribute_misc.hpp
+++ b/include/highfive/bits/H5Attribute_misc.hpp
@@ -30,7 +30,7 @@ namespace HighFive {
 
 inline std::string Attribute::getName() const {
     return details::get_name(
-        [&](char* buffer, hsize_t length) { return H5Aget_name(_hid, length, buffer); });
+        [&](char* buffer, size_t length) { return H5Aget_name(_hid, length, buffer); });
 }
 
 inline size_t Attribute::getStorageSize() const {

--- a/include/highfive/bits/H5File_misc.hpp
+++ b/include/highfive/bits/H5File_misc.hpp
@@ -80,7 +80,7 @@ inline File::File(const std::string& filename,
 inline const std::string& File::getName() const noexcept {
     if (_filename.empty()) {
         _filename = details::get_name(
-            [this](char* buffer, hsize_t length) { return H5Fget_name(getId(), buffer, length); });
+            [this](char* buffer, size_t length) { return H5Fget_name(getId(), buffer, length); });
     }
     return _filename;
 }

--- a/include/highfive/bits/H5Node_traits_misc.hpp
+++ b/include/highfive/bits/H5Node_traits_misc.hpp
@@ -166,7 +166,7 @@ inline size_t NodeTraits<Derivate>::getNumberObjects() const {
 
 template <typename Derivate>
 inline std::string NodeTraits<Derivate>::getObjectName(size_t index) const {
-    return details::get_name([&](char* buffer, hsize_t length) {
+    return details::get_name([&](char* buffer, size_t length) {
         return H5Lget_name_by_idx(static_cast<const Derivate*>(this)->getId(),
                                   ".",
                                   H5_INDEX_NAME,

--- a/include/highfive/bits/H5Path_traits_misc.hpp
+++ b/include/highfive/bits/H5Path_traits_misc.hpp
@@ -33,7 +33,7 @@ inline PathTraits<Derivate>::PathTraits() {
 
 template <typename Derivate>
 inline std::string PathTraits<Derivate>::getPath() const {
-    return details::get_name([this](char* buffer, hsize_t length) {
+    return details::get_name([this](char* buffer, size_t length) {
         return H5Iget_name(static_cast<const Derivate*>(this)->getId(), buffer, length);
     });
 }

--- a/include/highfive/bits/H5Reference_misc.hpp
+++ b/include/highfive/bits/H5Reference_misc.hpp
@@ -20,7 +20,7 @@ namespace HighFive {
 inline Reference::Reference(const Object& location, const Object& object)
     : parent_id(location.getId()) {
     obj_name = details::get_name(
-        [&](char* buffer, hsize_t length) { return H5Iget_name(object.getId(), buffer, length); });
+        [&](char* buffer, size_t length) { return H5Iget_name(object.getId(), buffer, length); });
 }
 
 inline void Reference::create_ref(hobj_ref_t* refptr) const {

--- a/include/highfive/bits/H5Utils.hpp
+++ b/include/highfive/bits/H5Utils.hpp
@@ -293,7 +293,7 @@ inline std::string get_name(T fct) {
         return std::string(buffer, length);
     }
     std::vector<char> bigBuffer(length + 1, 0);
-    fct(bigBuffer.data(), static_cast<hsize_t>(length) + 1);
+    fct(bigBuffer.data(), length + 1);
     return std::string(bigBuffer.data(), length);
 }
 

